File: problems/K Closest Points to Origin/kClosest.cs
public class Solution {
    public int[][] KClosest(int[][] points, int K) {
        var N = points.Length;
        var dists = new int[N];

        for (var i = 0; i < N; ++i)
            dists[i] = dist(points[i]);

        Array.Sort(dists);
        var distK = dists[K-1];

        var ans = new int[K][];
        var t = 0;

        for (int i = 0; i < N; ++i)
            if (dist(points[i]) <= distK)
                ans[t++] = points[i];

        return ans;
    }

    public int dist(int[] point) {
        return point[0] * point[0] + point[1] * point[1];
    }
}

File: problems/K Closest Points to Origin/kClosest.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int[]> maxHeap = new PriorityQueue<>((a,b)->b[0]*b[0]+b[1]*b[1]-a[0]*a[0]-a[1]*a[1]);
        
        for (int[] point : points) {
            maxHeap.add(point);
            
            if (k < maxHeap.size()) {
                maxHeap.poll();
            }
        }
        
        int[][] result = new int[maxHeap.size()][2];
        int idx = 0;
        
        while (!maxHeap.isEmpty()) {
            result[idx++] = maxHeap.poll();
        }
        
        return result;
    }
}
